lib/chkname.c

```diff
@@ -59,6 +63,21 @@ login_name_max_size(void)
 static bool
 is_valid_name(const char *name)
 {
+	if (streq(name, "")
+	 || streq(name, ".")
+	 || streq(name, "..")
+	 || strcaseeq(name, "none")
+	 || strcaseeq(name, "all")
+	 || strcaseeq(name, "except")
+	 || strprefix(name, "-")
+	 || strpbrk(name, " !\"#&*+,/:;@|")
```

Comment:
I've looked at the old Debian patches, and they also forbid a `~`.

I guess it's a good idea to also forbid that character, to avoid problems with for example

```sh
$ cd ~alx;
```

Cc: @zeha

Reply:
is that a problem for shadow?

Reply:
I think it's not a problem for shadow, but for core packages.

Some other characters in that list may not be a problem for shadow, but are for other core packages.